src/Microsoft.ML.Featurizers/CategoryImputer.cs

```diff
@@ -0,0 +1,766 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text;
+using Microsoft.ML;
+using Microsoft.ML.CommandLine;
+using Microsoft.ML.Data;
+using Microsoft.ML.EntryPoints;
+using Microsoft.ML.Featurizers;
+using Microsoft.ML.Internal.Utilities;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Transforms;
+using static Microsoft.ML.Featurizers.CommonExtensions;
+
+[assembly: LoadableClass(typeof(CategoryImputerTransformer), null, typeof(SignatureLoadModel),
+    CategoryImputerTransformer.UserName, CategoryImputerTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(IRowMapper), typeof(CategoryImputerTransformer), null, typeof(SignatureLoadRowMapper),
+   CategoryImputerTransformer.UserName, CategoryImputerTransformer.LoaderSignature)]
+
+[assembly: EntryPointModule(typeof(CategoryImputerEntrypoint))]
+
+namespace Microsoft.ML.Featurizers
+{
+    public static class CategoryImputerExtensionClass
+    {
+        /// <summary>
+        /// Create a <see cref="CategoryImputerEstimator"/>, which fills in the missing values in a column with the most frequent value. Supports Floats, Doubles, and Strings.
+        /// A string is assumed "missing" if it is empty.
```

Comment:
ML.NET already has a MissingValueReplacingEstimator with different replacement modes. What seems to be missing is a replacement mode for most frequent value. Wouldn't it be simpler to just add that mode to MissingValueReplacingEstimator?

Reply:
Yes, I agree this route of extending the existing transform. 

This is cleaner for the user as they don't have to learn a new transform, and there isn't two places to accomplish similar tasks. One transform increases discoverability and encourages the user to explore alternative imputation methods.

Reply:
That wont work unfortunately. To use this code you have to take a new dependency on the featurizers binary. The main ML.NET nuget cannot have any new dependencies added to it, so this has to be in a separate class/nuget package.

Reply:
ML.NET already has most of the plumbing in this specific case. There is already a MeanAggregatorOne class in MissingValueReplacingUtils.cs. Wouldn't it be simpler to add a ModeAggregratorOne and related classes and use the featurizer code only for the Onnx functionality?

---
In reply to: [363476040](https://github.com/dotnet/machinelearning/pull/4594#discussion_r363476040) [](ancestors = 363476040)

Reply:
As discussed offline this code is auto generated. The functionality will be migrated into the `MissingValueReplacingTransformer` over the next several weeks and this file will be removed, but its going in as is to unblock other development work.